crypto-extensions/src/main/java/net/corda/v5/crypto/extensions/DigestAlgorithm.java

```diff
@@ -0,0 +1,77 @@
+package net.corda.v5.crypto.extensions;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.InputStream;
+
+/**
+ * Digest extensions: Interface defining a custom digest calculation implementation. The interface should be implemented
+ * if a CPK developer wishes to provide support for digest algorithms beyond those supported by the Corda platform.
+ * The implementation of the interface must be coupled with the implementation of the [DigestAlgorithmFactory] which
+ * will be used to create the instances. For each algorithm there must be matching a pair of
+ * [DigestAlgorithmFactory] and [DigestAlgorithm] implementations.
+ * <p>
+ * Example of the algorithm and the factory implementations:
+ * <p>
+ * ```kotlin
```

Comment:
Not a change for this PR, but maybe we should raise a JIRA for adding a JAVA example here?

Reply:
Feel free if you like, I'm not concerned enough to raise a ticket.